access/types/stdasync.py

```diff
@@ -298,6 +299,21 @@ def _acceptSubmission(request, course, exercise, post_url, sdir: SubmissionDir):
     uids = get_uid(request)
     attempt = int(request.GET.get("ordinal_number", 1))
 
+    # Write LTI parameters for use in grading.
+    # The request is already authenticated and LTI signature is ignored.
+    if "lti" in exercise:
+        sdir.write_file("lti.json", json.dumps({
+            k: request.POST.get(k)
+            for k in (
+                "user_id",
+                "custom_student_id",
+                "lis_person_name_full",
+                "lis_person_name_given",
+                "lis_person_name_family",
+                "lis_person_contact_email_primary",
```

Comment:
Not a special reason but we do call the file `lti.json` so the names lti defines are logical. The way to switch this on is to enable LTI rather than "include personal information" so I would be hesitant to magically write "student_data.json" with custom values – I think there would just be more to wonder at some later time. This is not a strong opinion.